pkg/kubelet/util/util_windows_test.go

```diff
@@ -90,3 +96,103 @@ func TestParseEndpoint(t *testing.T) {
 	}
 
 }
+
+func TestIsUnixDomainSocket(t *testing.T) {
+	tests := []struct {
+		label          string
+		listenOnSocket bool
+		regularFile    bool
+		expectSocket   bool
+		expectError    bool
+		invalidFile    bool
+		listenOnPipe   bool
+	}{
+		{
+			label:          "Domain Socket File",
+			listenOnSocket: true,
+			expectSocket:   true,
+			expectError:    false,
+		},
+		{
+			label:        "Non Existent File",
+			invalidFile:  true,
+			expectSocket: false,
+			expectError:  true,
+		},
+		{
+			label:        "Regular File",
+			regularFile:  true,
+			expectSocket: false,
+			expectError:  false,
+		},
+		{
+			label:        "Named Pipe",
+			listenOnPipe: true,
+			expectError:  true,
+			expectSocket: false,
+		},
+	}
+	for _, test := range tests {
```

Comment:
Hmm.....I think there's too many if/else blocks in the test and the code reuse is not so high... Maybe it'd be easier to write separate test function for each test case?

Reply:
Done